src/GlobalsAttributeReader.php

```diff
@@ -0,0 +1,119 @@
+<?php
+declare(strict_types=1);
+
+namespace Zalas\PHPUnit\Globals;
+
+use PHPUnit\Event\Code\TestMethod;
+use PHPUnit\Event\Test\Prepared;
+use PHPUnit\Event\Test\PreparedSubscriber;
+use Zalas\PHPUnit\Globals\Attribute\Env;
+use Zalas\PHPUnit\Globals\Attribute\Putenv;
+use Zalas\PHPUnit\Globals\Attribute\Server;
+
+final class GlobalsAttributeReader implements PreparedSubscriber
+{
+    public function notify(Prepared $event): void
+    {
+        $this->readGlobalAttributes($event->test());
+    }
+
+    private function readGlobalAttributes(TestMethod $method)
+    {
+        $setVars = $this->findSetVarAttributes($method);
+
+        foreach ($setVars as $var) {
+            match (true) {
+                $var instanceof Env => $_ENV[$var->name] = $var->value,
+                $var instanceof Server => $_SERVER[$var->name] = $var->value,
+                $var instanceof Putenv => \putenv(\sprintf('%s=%s', $var->name, $var->value))
+            };
+        }
+
+        $unsetVars = $this->findUnsetVarAttributes($method);
+
+        foreach ($unsetVars as $var) {
+            $callable = match (true) {
+                $var instanceof Env => static function ($var) {
+                    unset($_ENV[$var->name]);
+                },
+                $var instanceof Server => static function ($var) {
+                    unset($_SERVER[$var->name]);
+                },
+                $var instanceof Putenv => static function ($var) {
+                    \putenv($var->name);
+                }
+            };
+
+            $callable($var);
+        }
+    }
+
+    private function findSetVarAttributes(TestMethod $method): array
+    {
+        $attributes = \array_filter(
+            $this->parseTestMethodAttributes($method),
+            static fn (Env|Server|Putenv $attribute) => false === $attribute->unset,
+            ARRAY_FILTER_USE_BOTH
+        );
+
+        \usort($attributes, static fn (Env|Server|Putenv $a, Env|Server|Putenv $b) => $a->getTarget() <=> $b->getTarget());
+
+        return $attributes;
+    }
+
+    private function findUnsetVarAttributes(TestMethod $method): array
+    {
+        $attributes = \array_filter(
+            $this->parseTestMethodAttributes($method),
+            static fn (Env|Server|Putenv $attribute) => true === $attribute->unset,
+            ARRAY_FILTER_USE_BOTH
+        );
+
+        \usort($attributes, static fn (Env|Server|Putenv $a, Env|Server|Putenv $b) => $a->getTarget() <=> $b->getTarget());
+
+        return $attributes;
+    }
+
+    /**
+     * @return array<Env|Putenv|Server>
+     */
+    private function parseTestMethodAttributes(TestMethod $method): array
```

Comment:
This method is currently called twice. I wonder what's the performance impact, but this is something that can be dealt with later.

Reply:
Called once now.